roxie/udplib/udptrr.cpp

```diff
@@ -949,9 +953,12 @@ class CReceiveManager : implements IReceiveManager, public CInterface
     virtual IMessageCollator *createMessageCollator(IRowManager *rowManager, ruid_t ruid)
     {
         IMessageCollator *msgColl = createCMessageCollator(rowManager, ruid);
-        if (udpTraceLevel >= 2) DBGLOG("UdpReceiver: createMessageCollator %p %u", msgColl, ruid);
-        SpinBlock b(collatorsLock);
-        collators[ruid] = msgColl;
+        if (udpTraceLevel >= 2)
+            DBGLOG("UdpReceiver: createMessageCollator %p %u", msgColl, ruid);
+        {
+            SpinBlock b(collatorsLock);
+            collators[ruid] = msgColl;
+        }
         msgColl->Link();
```

Comment:
Is this safe after the SpinBlock?

Reply:
I think it is, as the object is still owned by the caller.